src/Data/List/Relation/Binary/Permutation/Setoid.agda

```diff
@@ -44,44 +41,66 @@ _↭_ = Homogeneous.Permutation _≈_
 ------------------------------------------------------------------------
 -- Constructor aliases
 
+-- Constructor alias
+
+↭-pointwise : _≋_ ⇒ _↭_
+↭-pointwise = refl
+
 -- These provide aliases for `swap` and `prep` when the elements being
 -- swapped or prepended are propositionally equal
 
 ↭-prep : ∀ x {xs ys} → xs ↭ ys → x ∷ xs ↭ x ∷ ys
-↭-prep x xs↭ys = prep Eq.refl xs↭ys
+↭-prep x xs↭ys = prep ≈.refl xs↭ys
 
 ↭-swap : ∀ x y {xs ys} → xs ↭ ys → x ∷ y ∷ xs ↭ y ∷ x ∷ ys
-↭-swap x y xs↭ys = swap Eq.refl Eq.refl xs↭ys
-
-------------------------------------------------------------------------
--- Functions over permutations
-
-steps : ∀ {xs ys} → xs ↭ ys → ℕ
-steps (refl _)            = 1
-steps (prep _ xs↭ys)      = suc (steps xs↭ys)
-steps (swap _ _ xs↭ys)    = suc (steps xs↭ys)
-steps (trans xs↭ys ys↭zs) = steps xs↭ys + steps ys↭zs
+↭-swap x y xs↭ys = swap ≈.refl ≈.refl xs↭ys
 
 ------------------------------------------------------------------------
 -- _↭_ is an equivalence
 
 ↭-reflexive : _≡_ ⇒ _↭_
-↭-reflexive refl = refl (Pointwise.refl Eq.refl)
+↭-reflexive refl = ↭-pointwise ≋-refl
 
 ↭-refl : Reflexive _↭_
 ↭-refl = ↭-reflexive refl
 
 ↭-sym : Symmetric _↭_
-↭-sym = Homogeneous.sym Eq.sym
+↭-sym = Homogeneous.sym ≈.sym
+
+↭-transˡ-≋ : LeftTrans _≋_ _↭_
+↭-transˡ-≋ xs≋ys               (refl ys≋zs)
+  = refl (≋-trans xs≋ys ys≋zs)
+↭-transˡ-≋ (x≈y ∷ xs≋ys)       (prep y≈z ys↭zs)
+  = prep (≈.trans x≈y y≈z) (↭-transˡ-≋ xs≋ys ys↭zs)
+↭-transˡ-≋ (x≈y ∷ w≈z ∷ xs≋ys) (swap eq₁ eq₂ ys↭zs)
+  = swap (≈.trans x≈y eq₁) (≈.trans w≈z eq₂) (↭-transˡ-≋ xs≋ys ys↭zs)
+↭-transˡ-≋ xs≋ys               (trans ys↭ws ws↭zs)
+  = trans (↭-transˡ-≋ xs≋ys ys↭ws) ws↭zs
+
+↭-transʳ-≋ : RightTrans _↭_ _≋_
+↭-transʳ-≋ (refl xs≋ys)         ys≋zs
+  = refl (≋-trans xs≋ys ys≋zs)
+↭-transʳ-≋ (prep x≈y xs↭ys)     (y≈z ∷ ys≋zs)
+  = prep (≈.trans x≈y y≈z) (↭-transʳ-≋ xs↭ys ys≋zs)
+↭-transʳ-≋ (swap eq₁ eq₂ xs↭ys) (x≈w ∷ y≈z ∷ ys≋zs)
+  = swap (≈.trans eq₁ y≈z) (≈.trans eq₂ x≈w) (↭-transʳ-≋ xs↭ys ys≋zs)
+↭-transʳ-≋ (trans xs↭ws ws↭ys)  ys≋zs
+  = trans xs↭ws (↭-transʳ-≋ ws↭ys ys≋zs)
 
 ↭-trans : Transitive _↭_
-↭-trans = trans
+↭-trans (refl xs≋ys) ys↭zs = ↭-transˡ-≋ xs≋ys ys↭zs
+↭-trans xs↭ys (refl ys≋zs) = ↭-transʳ-≋ xs↭ys ys≋zs
+↭-trans xs↭ys ys↭zs        = trans xs↭ys ys↭zs
```

Comment:
This is a big change in behaviour of `trans` and may break user's proofs? Regardless there should be documentation here explaining why we're using this non-obvious definition?

Reply:
Interesting!
... yes, the intentional behaviour does change, but exactly by analogy with the corresponding version in `Propositional`. But if you *really* want the intentional behaviour, you still have access to the vanilla `trans` constructor... is that (not) good enough?

Reply:
DONE, as `↭-trans′`...

Reply:
Mind you, a proof I couldn't be bothered to investigate is whether the `steps` properties hold for *this* version of 'smart' `trans`... which I think in fact should be provable, given the way pointwise equality is being pushed through the derivation tree to the leaves... but as I say, that's work I *really* don't want to do.

Reply:
Out of interest, does the introduction of `↭-trans′`  break any of *your* proofs?

Reply:
Forgot to add the commentary you requested. Probably a bit verbose now!